packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -1306,8 +1327,28 @@ export class IntervalCollection<TInterval extends ISerializableInterval>
 			throw new LoggingError("Change API requires an ID that is a string");
 		}
 
+		const startIndex = this.getIndex(start);
+		const endIndex = this.getIndex(end);
+		if (startIndex !== undefined && endIndex !== undefined && startIndex > endIndex) {
+			throw new UsageError("interval start cannot be greater than end");
+		}
+
 		const interval = this.getIntervalById(id);
 		if (interval) {
+			if (
+				startIndex !== undefined &&
+				startIndex === endIndex &&
+				interval instanceof SequenceInterval
+			) {
+				const newStartSide = typeof start === "object" ? start.side : interval.startSide;
+				const newEndSide = typeof end === "object" ? end.side : interval.endSide;
+				if (newStartSide < newEndSide) {
```

Comment:
Yeah, it's not what I was expecting either. It might be more readable to have `newStartSide===Side.Before && newEndSide===Side.After`. Side refers to whether that interval endpoint boundary will move left (content is After) or right (content is Before) of newly inserted content at that position.

Reply:
Ah ok, I think either that or changing the error message to avoid saying "greater than" (since the Side.After enum value is technically greater than that of Side.Before) would help with clarification.

Reply:
I was confused by the error message here as well. If both endpoints are on the same character and the start endpoint has `Side.Before`, while the end endpoint has `Side.After`, the start side would _not_ be greater than the end side.

If that character got deleted, then the endpoints would likely end up in a state in which the start is past the end, but in its current state it would be a valid interval containing a single character.

I suppose:
1. Why do we want to consider such an interval invalid?
2. Why do we not want to consider an interval in which the start side is `Side.After` and the end side is `Side.Before` invalid? This is a case in which the start position _is_ greater than the end position.

There's some docs on stickiness [here](https://github.com/microsoft/FluidFramework/blob/next/packages/dds/sequence/README.md#interval-stickiness). I'm curious if there are thoughts on how to update them to make the concept of "sides" more intuitive. I think they greatly simplify the public API, but perhaps can make the internal implementation more complex.

Reply:
Yes, I was confused: I thought the endpoint ref would actually move when content was inserted. I think I finally have it correct now where having startSide > endSide at the same position is disallowed.

Changes to documentation to make it clearer: #17383